### PR TITLE
CTCTRADERS-400 

### PR DESCRIPTION
**Pull request by amaalali**
Limit the adding of 007 to existing movement to only when there are failed 007 messages

### CODE REVIEW

File: app/controllers/MovementsController.scala [hunk 1]
```diff
@@ -56,10 +58,13 @@ class MovementsController @Inject()(
 )(implicit ec: ExecutionContext)
     extends BackendController(cc) {
 
+  private val atLeastOneMessageSent: Seq[MovementMessage] => Boolean =
+    _.exists(x => x.optStatus.contains(SubmissionSucceeded) || x.optStatus.isEmpty)
```
Comment (jhkldev) [inline app/controllers/MovementsController.scala:62]: Can the x.optStatus.isEmpty condition be removed?

Comment (amaalali) [reply app/controllers/MovementsController.scala:62]: No, if there is a response from upstream we can shortcircuit out of the list traversal

Comment (RobertLaverick) [reply app/controllers/MovementsController.scala:62]: Shouldn't the x.optStatus.isEmpty be first in this condition? I'm not sure how it'll handle contains if it is empty.

Comment (amaalali) [reply app/controllers/MovementsController.scala:62]: `None.contains` will return false

**Pull request merged**